src/java/org/apache/cassandra/cql3/QueryEvents.java

```diff
@@ -72,8 +82,9 @@ public void notifyQuerySuccess(CQLStatement statement,
     {
         try
         {
+
             for (Listener listener : listeners)
-                listener.querySuccess(statement, query, options, state, queryTime, response);
+                listener.querySuccess(statement, possiblyObfuscateQuery(statement, query), options, state, queryTime, response);
```

Comment:
@sumanth-pasupuleti would it be possible to move this method invocation _outside_ of `for`, because if you have 10 listeners, you are basically obfuscating the same string 10 times but 1 time is just enough. This holds for all other places in this class.

Reply:
Updating based on our offline conversation

- In most practical scenarios, one would either have no / one listener configured in which case, it is better we do obfuscation inside the forloop vs outside, since if it is outside, independent of whether there are any listeners, we do the obfuscation once.
- In case of > 1 listener, performance penalty of having multiple listeners is likely to exceed an additional round of obfuscation

Reply:
Another idea: 
Can we do this before for loop to avoid the overhead / unnecessary cost
`query = listeners.size()>0? possiblyObfuscateQuery(statement, query) : query;
       `

Reply:
@sumanth-pasupuleti feel free to implement, I ll leave this to your consideration.

Reply:
> Another idea:
> Can we do this before for loop to avoid the overhead / unnecessary cost
> `query = listeners.size()>0? possiblyObfuscateQuery(statement, query) : query; `

incorporating ^